### PR TITLE
Active active test fix

### LINKED ISSUE #854: Fix and reenable Active/Active functional tests
There are testsuite failures with A/A depoyments

### PR DESCRIPTION
**Pull request by ryanemerson**
Closes #854

The source of the original failures was fixed by https://github.com/keycloak/keycloak/pull/30841

However, when investigating the issue I realised that the testsuite was very flaky and failures often occurred when executing the `FailoverTest` in conjunction with the other tests. This PR reenables the active-active tests as well adding several fixes/improvements to the testsuite to reduce the chance of flaky failures.

### CODE REVIEW

**Review by ahus1 — APPROVED** [review]
LGTM, thank you for all these fixes to make it more stable!

**Review by mhajas — APPROVED** [review]
Thank you @ryanemerson!

**Review by mhajas — CHANGES_REQUESTED** [review]
We need to revert also this commit here: https://github.com/keycloak/keycloak-benchmark/commit/252559a6780736bedb9f4deb254f5765c6c80993

Sorry I forgot to mention it in the issue

File: .github/workflows/rosa-cluster-auto-provision-on-schedule.yml [hunk 1]
```diff
@@ -115,6 +115,14 @@ jobs:
...
+  run-functional-tests-active-active:
+    needs: keycloak-deploy-active-active
+    uses: ./.github/workflows/rosa-run-crossdc-func-tests.yml
+    with:
+      activeActive: true
+      clusterPrefix: gh-keycloak # ${{ env.CLUSTER_PREFIX }} -- unfortunately 'env.' doesn't work here
+    secrets: inherit
+
   run-scaling-benchmark-active-active:
     needs: keycloak-deploy-active-active
```
Comment (mhajas) [inline .github/workflows/rosa-cluster-auto-provision-on-schedule.yml:127]:
```suggestion
    needs: run-functional-tests-active-active
```

**Comment by ryanemerson** [comment]
Good spot @mhajas, I've accepted your suggestion.

**Review by mhajas — APPROVED** [review]
Thank you @ryanemerson

**Review by mhajas — APPROVED** [review]
Thanks @ryanemerson!

**Pull request merged**